index.js

```diff
@@ -0,0 +1,172 @@
+                  break;
+                      }
+            }
+            else {
+              _lastElementFocused.value = _lastElementFocused.value + this.innerText;
+            }
+          });
+          vkc.appendChild(button);
+          vkr.appendChild(vkc);
+          target.appendChild(vkr);
+        }
+      }
+    },
+    getMatrix: function(language,shift = 'num') {
+      var matrix = {
+        symbols: [['!','@','$','%','^','&','*','(',')','_','+']],
+        num: [['1','2','3','4','5','6','7','8','9','0']],
+        en: [
+        ['q','w','e','r','t','y','u','i','o','p'],
+        ['+tab','a','s','d','f','g','h','j','k','l'],
+        ['+shift','z','x','c','v','b','n','m','.',','],
+        ['+international','+space','+backspace','+enter']
+      ],
+        ru: [
+          ['й','ц','у','к','е','н','г','ш','щ','з','х','ъ'],
+          ['+tab','ф','ы','в','а','п','р','о','л','д','ж','э'],
+          ['+shift','я','ч','с','м','и','т','ь','б','ю','ё','.',','],
+          ['+international','+space','+backspace','+enter']
+        ]};
+      return matrix[shift].concat(matrix[language]);
+    },
+    init: function(args) {
+      if (args != undefined && args != null) {
+        if (Object.keys(args).length > 0) {
+          var owner = this;
+  
+          window._lastElementFocused = null;
+  
+          var target = document.getElementById(args['targetId']);
+          var language = args['defaultLanguage'];
+          var elements = document.querySelectorAll(args['inputSelector']);
+  
+          _lastElementFocused = elements[0];
+  
+          for (var i = 0; i < elements.length; i++) {
+            elements[i].addEventListener('focus', function () {
+              _lastElementFocused = this;
+            });
+          }
+          owner.generate(target,owner.getMatrix(language), language);
+        }
+      }
+    }
+  }
+  let myStorage = window.localStorage; 
+  let Keyboard =  document.createElement('div');
+  let Input = document.createElement('textarea');
+  let defLang = myStorage.getItem('language');
+  Input.setAttribute('data-virtual-element','');
+  Input.setAttribute('cols','30');
+  Input.setAttribute('rows','10');
+  Keyboard.setAttribute('id','virtual-keyboard');
+  document.getElementsByTagName('body')[0].appendChild(Input);
+  document.getElementsByTagName('body')[0].appendChild(Keyboard);
+  VirtualKeyboard.init({targetId: 'virtual-keyboard', defaultLanguage: defLang == null ? 'en' : defLang, inputSelector: '[data-virtual-element]'});
+ 
+  document.addEventListener('keydown',(event)=>{
+	if(event.key == "Tab") event.preventDefault();
+    let buttons = document.getElementsByClassName('virtual-keyboard-button');
+    for(let i =0; i<buttons.length;i++){
+        if( buttons[i].innerHTML == (event.key == ' ' ? 'Space': event.key)){
+            buttons[i].style = "background-color: white";
+            setTimeout(()=>{buttons[i].removeAttribute('style');},500);
```

Comment:
why do you use setTimeout here but not native handlers such as 'keyup'?

Reply:
still left setInterval (highlighting interactive buttons might be realized via adding/removing CSS class to DOM element)
and code formating still not good (in for loops and if operators)